cmd/metrics-v2.go

```diff
@@ -1598,47 +1598,50 @@ func getGoMetrics() *MetricsGroup {
 	return mg
 }
 
+// getHistogramMetrics fetches histogram metrics and returns it in a []Metric
+// Note: Typically used in MetricGroup.RegisterRead
+func getHistogramMetrics(hist *prometheus.HistogramVec, desc MetricDescription) []Metric {
+	ch := make(chan prometheus.Metric)
+	go func() {
+		defer close(ch)
+		// Collects prometheus metrics from hist and sends it over ch
+		hist.Collect(ch)
+	}()
+
+	// Converts metrics received into internal []Metric type
+	var metrics []Metric
+	for promMetric := range ch {
+		dtoMetric := &dto.Metric{}
+		err := promMetric.Write(dtoMetric)
+		if err != nil {
+			logger.LogIf(GlobalContext, err)
```

Comment:
will we have a leaked goroutine here if we have this error ? (I am not sure if this error out in the first place)

Reply:
@vadmeste yes, good catch! 
@harshavardhana yes, I think that's a simple solution for an extremely rare case. I shall update my PR to address this.